### PR TITLE
Add Cannock Chase, UK

### LINKED ISSUE #2526: [Source Request]: Cannock Chase, UK
### Municipality / Region

Cannock Chase, UK

### Collection Calendar Webpage

https://www.cannockchasedc.gov.uk/residents/recycling-waste/waste-collection-check-your-dates

### Example Address

25 Brereton Road, RUGELEY, WS15 1DN (has garden waste collection)
27 Brereton Road, RUGELEY, WS15 1DN (does not have garden waste collection)

### Collection Data Format

As html on a webpage, Something different (add to additional information)

### Additional Information

API is undocumented but you can see the REST calls being made.

### PR DESCRIPTION
**Pull request by marcjay**
- Added Cannock Chase, UK source
- Fixed typos in Contributing guide
- Fixed typo in hastings_gov_uk source
- Fix filename typo

Resolves #2526

```bash
$ ./test_sources.py -s cannock_chase_dc_gov_uk -l -i
Testing source cannock_chase_dc_gov_uk ...
  found 6 entries for Test_001
    2024-10-17 : Garden waste [mdi:leaf]
    2024-10-17 : Recycling [mdi:recycle]
    2024-10-24 : Refuse [mdi:trash-can]
    2024-10-31 : Garden waste [mdi:leaf]
    2024-10-31 : Recycling [mdi:recycle]
    2024-11-07 : Refuse [mdi:trash-can]
  found 4 entries for Test_002
    2024-10-17 : Recycling [mdi:recycle]
    2024-10-24 : Refuse [mdi:trash-can]
    2024-10-31 : Recycling [mdi:recycle]
    2024-11-07 : Refuse [mdi:trash-can]
  found 7 entries for Test_003
    2024-10-14 : Garden waste [mdi:leaf]
    2024-10-21 : Recycling [mdi:recycle]
    2024-10-21 : Refuse [mdi:trash-can]
    2024-10-28 : Garden waste [mdi:leaf]
    2024-11-04 : Recycling [mdi:recycle]
    2024-11-04 : Refuse [mdi:trash-can]
    2024-11-11 : Garden waste [mdi:leaf]
  found 5 entries for Test_004
    2024-10-16 : Recycling [mdi:recycle]
    2024-10-23 : Refuse [mdi:trash-can]
    2024-10-30 : Recycling [mdi:recycle]
    2024-11-06 : Refuse [mdi:trash-can]
    2024-11-13 : Recycling [mdi:recycle]
```

### CODE REVIEW

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/cannock_chase_dc_gov_uk.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            "Method": "CollectionDates",
+            "UPRN": self._uprn,
+            "Postcode": self._postcode,
+        }
+
+        r = requests.post(API_URL, data=args)
+        r.raise_for_status()
+
+        ns = {"ws": "http://webservices.whitespacews.com/"}
+        tree = ET.fromstring(r.text)
```
Comment (marcjay) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/cannock_chase_dc_gov_uk.py:58]: I'm noting here that the `bandit` hook flagged:
> cannock_chase_dc_gov_uk.py:58: B314[bandit]: MEDIUM: Using xml.etree.ElementTree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace xml.etree.ElementTree.fromstring with its defusedxml equivalent function or make sure defusedxml.defuse_stdlib() is called

I was hesitant to add in a new dependency for `defusedxml` and can see a few existing uses of `xml.etree.ElementTree.fromstring`, but if you're ok with it, I can add it and replace all usages of `xml.etree.ElementTree.fromstring`?

Comment (5ila5) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/cannock_chase_dc_gov_uk.py:58]: I don't think it's that big of a security concern here.

**Comment by 5ila5** [comment]
Could you please resolve the mypy errors (Item "None" of "Element | None" has no attribute "text") at the `....find("...").text` lines

maybe replace `SERVICE_NAME_MAP.get(service)` with `SERVICE_NAME_MAP.get(service, service)` so that there still is a name when they add a new one or change a name

**Comment by marcjay** [comment]
Thank you for the review, that all made sense, have made the changes

**Comment by 5ila5** [comment]
Thanks for your contribution

**Pull request merged**